### ISSUE TITLE
Add an option to not unpack sources.

### ISSUE THREAD

**Issue by Keno** [JuliaPackaging/BinaryBuilder.jl#308]
On occasion, it can be useful to not unpack a src tarball (e.g. because the tarball has a bunch of data files that need to extracted to a specific directory). Add an option to (not) do that.

**Comment by staticfloat** [comment]
Is it not better to just move the extracted files after they've already been extracted?  Or are you saying you'd only want to extract a small portion of the files within the tarball?

**Comment by Keno** [comment]
My problem is that I don't know what the extracted files are if they get extracted to the top level directory all at once.

**Comment by staticfloat** [comment]
Your tarball doesn't contain a containing directory? 

![how wude!](https://media.giphy.com/media/3owzWfuEMmNX4rquJ2/giphy.gif)

**Comment by Keno** [comment]
Yeah, I know :/

**Comment by giordano** [comment]
Fixed by #643 and #671: now users can either
* use
  ```julia
  ArchiveSource(...; unpack_target = "subdir")
  ```
  to unpack the tarbomb to `/workspace/srcdir/subdir`
* or use
  ```julia
  FileSource(...)
  ```
  and the archive will simply not be unpacked
